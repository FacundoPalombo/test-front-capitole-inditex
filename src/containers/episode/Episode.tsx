import React from 'react';
import { useParams } from 'react-router-dom';
import { useIsFetching, useQuery } from '@tanstack/react-query';

import { episodes as episodesQuery } from '../../queries/podcasts';

import EpisodeDetail from './components/EpisodeDetail/EpisodeDetail';
import Skeleton from './components/EpisodeDetailSkeleton/EpisodeDetailSkeleton';

const Episode = () => {
  const params = useParams();
  const { data: podcasts } = useQuery(episodesQuery(params.podcastId));

  const isLoading = useIsFetching() > 0;

  const podcast = podcasts?.results?.find(
    (podcast) => podcast?.trackId?.toString() === params.episodeId
  );

  if (isLoading) return <Skeleton />;
  return (
    podcast && (
      <EpisodeDetail
        description={podcast.description!}
        episodeUrl={podcast.episodeUrl!}
        trackName={podcast.trackName!}
      />
    )
  );
};

export default Episode;
