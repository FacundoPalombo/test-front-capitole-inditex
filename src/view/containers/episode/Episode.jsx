import React from 'react';
import { Await, useRouteLoaderData } from 'react-router-dom';
import styles from './styles.module.scss';

const Episode = () => {
  const { podcasts } = useRouteLoaderData('podcasts');
  console.log(podcasts);

  function renderComponent(podcasts) {
    return (
      <div className={styles.episode__content}>
        <h2 className={styles.episode__title}>{podcasts.trackName}</h2>
        <code className={styles.episode__description}>
          {podcasts.description}
        </code>
        <audio
          className={styles.episode__audio}
          src={podcasts.episodeUrl}
          controls
        ></audio>
      </div>
    );
  }

  return (
    <section id="episode" className={styles.episode}>
      <React.Suspense fallback={<div>loading...</div>}>
        <Await resolve={podcasts} errorElement={<div>ups error</div>}>
          {renderComponent}
        </Await>
      </React.Suspense>
    </section>
  );
};

export default Episode;
