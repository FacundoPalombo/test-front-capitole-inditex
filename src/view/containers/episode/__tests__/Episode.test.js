import React from 'react';
import Episode from '../Episode';
import { render } from '@testing-library/react';

describe('Episode unit test', () => {
  it('should render ok with props', () => {
    const { asFragment } = render(<Episode />);
    expect(asFragment()).toMatchSnapshot();
  });
});
