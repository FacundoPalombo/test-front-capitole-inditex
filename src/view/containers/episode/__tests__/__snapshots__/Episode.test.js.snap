// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episode unit test should render ok with props 1`] = `
<DocumentFragment>
  <section
    id="episode"
  >
    Episode
  </section>
</DocumentFragment>
`;
